sunpy/map/maputils.py

```diff
@@ -124,6 +124,17 @@ def sample_at_coords(smap, coordinates):
     return smap.data[smap.wcs.world_to_array_index(coordinates)]
 
 
+def _edge_coordinates(smap):
+    # We need to strip the units from edges before handing it to np.concatenate,
+    # as the .unit attribute is not propagated in np.concatenate for numpy<1.17
+    # When sunpy depends on numpy>=1.17 this unit replacing code can be removed
+    edges = [edge.to_value(u.pix) for edge in map_edges(smap)]
+    # Calculate all the edge pixels
+    edge_pixels = np.concatenate(edges) * u.pix
```

Comment:
It's a marginal improvement, but seeing how this PR is about perf, we might as well not copy the array one more time.
```suggestion
    edge_pixels = u.Quantity(np.concatenate(edges), unit=u.pix, copy=False)
```